Dockerfile

```diff
@@ -1,14 +1,20 @@
 FROM node:stretch-slim
 
-RUN apt update && apt install -y git openssh-server ca-certificates openssl jq gettext xmlstarlet build-essential
-
-RUN npm install sfdx-cli prettier prettier-plugin-apex --global
-RUN sfdx --version
-RUN sfdx plugins --core
-
-# Work-around to make java installation work
-# as per https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=863199
-RUN mkdir -p /usr/share/man/man1
-RUN apt-get install openjdk-8-jdk -y
+RUN apt update && apt install -y git openssh-server ca-certificates openssl jq gettext xmlstarlet build-essential libcurl4-gnutls-dev libexpat1-dev libz-dev libssl-dev make && \
+    git clone git://git.kernel.org/pub/scm/git/git.git && \
```

Comment:
Why do we need to update git version for a docker image?

Reply:
the version if git is 2.11 and i would like to use the option -o
https://git-scm.com/docs/git-push/2.11.1
https://git-scm.com/docs/git-push/

as you can see the version 2.11 wasn't included.

Reply:
You shouldn't depend on the git version from this image.
As far as I know this git included here is just a dependency from npm and may not be available on the future.

If you wanna run something that have as the main command git then you should have your own image for that. So you can release versions for that new image.

In GitLab CI/CD Pipeline it's possible to select a different image per job: https://docs.gitlab.com/ee/ci/yaml/#image